src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,20 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (MessageWebSocketClientCertificateSupported)
+            {
+                // Use the first app-provided client cert that can be successfully converted into a WinRT client cert (if any).
+                foreach (X509Certificate dotNetClientCert in options.ClientCertificates)
+                {
+                    RTCertificate winRtClientCert = ConvertDotNetClientCertToWinRtClientCert(dotNetClientCert);
+                    if (winRtClientCert != null)
```

Comment:
Throwing an exception here seems less controversial to me. Unlike the other comment, this exception would be related to the app's certificate usage (fixable by the app developer) instead of the Windows build number (nothing the app developer can do about it).